src/Microsoft.OData.Core/UriParser/Binders/LiteralBinder.cs

```diff
@@ -22,33 +22,46 @@ internal static QueryNode BindLiteral(LiteralToken literalToken)
         {
             ExceptionUtils.CheckArgumentNotNull(literalToken, "literalToken");
 
+            if (!string.IsNullOrEmpty(literalToken.OriginalText))

```

Comment:
>if (!string.IsNullOrEmpty(literalToken.OriginalText)) [](start = 12, length = 53)

This means that we won't set the type reference if the originalText is null.  Is that the desired behavior? Would we ever know the typereference but not the originaltext?

Reply:
We will set the type reference even if the OriginalText is null but by getting the type of the value from Reflection. 

None of our tests have the scenario where we have the TypeReference but not the OriginalText. We will keep the original behavior, because it is likely either it is not a valid case or a very obscured case since there are no known issues around this.